### DISCUSSION TITLE
手动部署v1.8.5升级到v1.10.0时本地账号认证登录时出现 500 Internal Server Error

### DISCUSSION THREAD

**Discussion by gdzw** [hhyo/Archery#2455] [Q&A]
**手动部署v1.8.5升级到v1.10.0按照如下升级步骤进行升级：**
1、下载对应 Release 版本的压缩包: Releases
2、按照配置文件的说明修改相关配置: 修改配置
3、重新安装依赖, 以免存在依赖变更pip install -r requirements.txt
4、应用数据库变更 python manage.py migrate
5、重新启动服务: 启动

当前v1.8.5未启用任何外部认证，账号为本地认证的方式，升级到v1.10.0后，页面上输入账号/密码登录，报如下错误：

Django Version: 4.1.10
Python Version: 3.9.10
OS Version: CentOS Linux release 7.9.2009

Request Method: POST
Request URL: http://192.168.13.49:80/authenticate/

Traceback (most recent call last):
  File "/data/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 56, in inner
    response = get_response(request)
  File "/data/venv4archery/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/data/archery1.10.0/common/auth.py", line 118, in authenticate_entry
    result = new_auth.authenticate()
  File "/data/archery1.10.0/common/auth.py", line 84, in authenticate
    lock_count = int(self.sys_config.get("lock_cnt_threshold", 5))

Exception Type: ValueError at /authenticate/
Exception Value: invalid literal for int() with base 10: 'X5u0cP5cKotxfutQNIFDYQ=='

/data/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py, line 56, in inner
        return inner
    else:
        @wraps(get_response)
        def inner(request):
            try:
                response = get_response(request) …
            except Exception as exc:
                response = response_for_exception(request, exc)
            return response
        return inner
Local vars
Variable	Value
exc	
ValueError("invalid literal for int() with base 10: 'X5u0cP5cKotxfutQNIFDYQ=='")
get_response	
<bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x7f00e88d3d90>>
request	
<WSGIRequest: POST '/authenticate/'>

**Comment by LeoQuote** [comment]
建议使用项目带的 sql 语句进行升级改表, migration 的方式不支持的.

**Reply by gdzw** [reply]
(venv4archery) [root@sql-test archery1.10.0]# python manage.py dbshell < src/init_sql/v1.9.0.sql
import local settings failed, ignored
(venv4archery) [root@sql-test  archery1.10.0]# python manage.py dbshell < src/init_sql/v1.10.0.sql
import local settings failed, ignored

(venv4archery) [root@sql-test archery1.10.0]# bash startup.sh 
import local settings failed, ignored
Unlinking stale socket /data/archery1.10.0/supervisor.sock

应用了对应的升级SQL，依旧是报相同的错误

**Reply by gdzw** [reply]
当前定位到问题，为`archery`.`sql_config`表，在File "/data/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 56, in inner 无法将varchar类似的值转换为int类型：

`item` = 'lock_cnt_threshold', `value` = 'X5u0cP5cKotxfutQNIFDYQ=='
`item` = 'lock_time_threshold', `value` = 'hU0qtVlvCnpD4_107CsDsw=='
调整为：
`item` = 'lock_cnt_threshold', `value` = '0'
`item` = 'lock_time_threshold', `value` = '0'

之后可以验证登录，但在登录时，跳转到http://192.168.13.49/login/2fa/ ，并没有开启2fa认证，这一步如何取消这个页面的跳转？

**Reply by LeoQuote** [reply]
可以把 sql_config 备份后清空，然后再逐个恢复

**Reply by gdzw** [reply]
清空 sql_config 的方式是可以，但对于升级来说，value都不是明文，感觉还是不够友好的

**Reply by gdzw** [reply]
还有sql_instance 表的user列也是没法解析出来，也需要手动替换？

**Reply by LeoQuote** [reply]
我比较怀疑是你 secretKey 没有手动替换, 使用了我们项目里的 secret key, 你可以找一下 1.8.5 之前的 secretkey 是什么, 然后换回老的 secret key , 看能否解决问题.

**Reply by gdzw** [reply]
1.8.5版本也是使用的项目里面的SECRET_KEY = 'hfusaf2m4ot#7)fkw#di2bu6(cv0@opwmafx5n#6=3d%x^hpl6'

**Reply by gdzw** [reply]
我是将现在的机器，克隆出新的一套出来，进行升级操作的，不知secret key在不同机器上是否有关？
